compiler/noirc_driver/src/program.rs

```diff
@@ -27,4 +27,6 @@ pub struct CompiledProgram {
     pub debug: DebugInfo,
     pub file_map: BTreeMap<FileId, DebugFile>,
     pub warnings: Vec<SsaReport>,
+    /// Names of the functions in the program. These are used for more informative debugging and benchmarking.
+    pub names: Vec<String>,
```

Comment:
Same question r.e. making the names 1-to-1 with the functions

Reply:
We perform our ACIR codegen for the backend [using the `Program` structure](https://github.com/noir-lang/noir/blob/f21129ef05efb76c5df6ee15a134f1ea535d8e90/acvm-repo/acir/src/lib.rs#L62) as this is what ultimately gets sent to a backend. It would be slightly separate work and a larger refactor to start including debug information as part of the `Program` as I don't think we should send debug info such as names to the backend. If the codegen is correct, there should be a name associated with each generated acir as part of `SsaCircuitArtifact`.